app/components/Videos/Upload/FileUpload.tsx

```
@@ -25,51 +21,125 @@ const truncateUri = (uri: string): string => {
+  // Filter out null results and combine translations
+  const translations = await Promise.all(translationPromises);
+  const languageTranslations = translations.filter(Boolean);
+
+  console.log('translations:', translations);
+  console.log('Language translations:', languageTranslations);
+
+  // Merge translations efficiently
+  return languageTranslations.filter(
+    (translation): translation is { [key: string]: Chunk[] } => translation !== null)
+      .reduce((acc, curr) => ({
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)